### PR TITLE
[docs] mv airflow migration strategy to "Migration Guides" section

### PR DESCRIPTION
**Pull request by cmpadden**
## Summary & Motivation

- Separates Airflow migration strategy to separate "Migration Guides" section

image

Questions
- [x] ~Should "Migrations" be a sub-section in "Guides"~
- [x] ~Is [arrow-top-right-on-square](https://heroicons.com/) icon appropriate?~
- [x] ~What is ideal placement on sidebar?~
- [x] ~In screenshot you can see "Guides" is being highlighted despite "Migrating from Airflow" being selected; this is likely a bug that needs to be addressed~

*Edit* questions have been resolved, as section has been moved within "Guides" and an icon is no longer necessary.

### CODE REVIEW

**Comment by cmpadden** [comment]
* **#22446** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22446?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @cmpadden and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by cmpadden** [comment]
@erinkcochran87 - after our conversation I've simplified these changes slightly. The previous `Guides > Migrating to Dagster` has been restored, but the the `/integrations/airflow/migrating-to-dagster` page has still been removed.

image

Let me know if you have any other feedback, thank you!

**Review by yuhan — COMMENTED** [review]
image

(side note: the screenshot in pr summary seems outdated)

I'm a little wary that the "version migration" and "migrating to dagster" may lead to confusion. thoughts on changing "version migration" to "version upgrade"?

other than that, i think this is a great start to a) clarify migration vs integration, b) set up a strong content hierarchy for migration paths.

**Comment by cmpadden** [comment]
> image
> 
> (side note: the screenshot in pr summary seems outdated)
> 
> I'm a little wary that the "version migration" and "migrating to dagster" may lead to confusion. thoughts on changing "version migration" to "version upgrade"?
> 
> other than that, i think this is a great start to a) clarify migration vs integration, b) set up a strong content hierarchy for migration paths.

That's a good point, I think either way makes sense to me, so am happy to go with the consensus.

Though one thing to note is "Version migration" is at the `/migration` path, and not something like `/guides/version-migration`. I was going to originally create a `/migrations` but found that to conflict.

File: docs/content/guides/migrations.mdx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: Migrations | Dagster
```
Comment (erinkcochran87) [inline docs/content/guides/migrations.mdx:2]:
```suggestion
title: Migrating to Dagster guides | Dagster Docs
```

`title:` convention is `title_of_page | Dagster Docs`

**Review by erinkcochran87 — COMMENTED** [review]
Thanks for bearing with me on this, Colton! A few small things but looking good otherwise!

**Comment by cmpadden** [comment]
> Thanks for bearing with me on this, Colton! A few small things but looking good otherwise!

Thank you - I've commit your suggestions. 🙏

**Review by erinkcochran87 — APPROVED** [review]

**Comment by cmpadden** [comment]
### Merge activity

* **Jun 13, 3:55 PM EDT**: @cmpadden started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/22446).

**Pull request merged**